### ISSUE TITLE
Regression: Aborted (core dumped) when running .NET 6.0 app in Docker on linux-arm host

### ISSUE THREAD

**Issue by JustArchi** [dotnet/runtime#62402]
### Description

Hello,

One of the users of my app reported that after upgrade from .NET 5.0 to .NET 6.0, the application no longer works in Docker container. I worked with the user to provide as much info as possible.

Upon investigation, the process crashes with `Aborted (core dumped)` when ran with `dotnet app.dll`.

### Reproduction Steps

It's very hard for me to give reproduction steps as I'm unable to reproduce this myself. The user managed to reproduce this with my fixed Docker tag, so if you feel lucky you can give it a try, but chance is it'll work fine for you.

```
docker run -it --rm justarchi/archisteamfarm:5.2.0.10
```

Output of failure:

```
root@omv:~/nas_smb# docker run -it --rm justarchi/archisteamfarm:5.2.0.10
Unable to find image 'justarchi/archisteamfarm:5.2.0.10' locally
5.2.0.10: Pulling from justarchi/archisteamfarm
Digest: sha256:a759e758b2641ad57950b778eddea42380303e5af28bf137cd64d7615dce66b0
Status: Downloaded newer image for justarchi/archisteamfarm:5.2.0.10

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
Aborted
```

If the program boots, you weren't lucky, I wasn't :slightly_frowning_face:.

If by any chance you'd be interested in the Dockerfile the app was built for, you can find it **[here](https://github.com/JustArchiNET/ArchiSteamFarm/blob/5.2.0.10/Dockerfile)**, but I don't believe the app is at fault here.

### Expected behavior

The program runs in Docker container just as before in .NET 5.0.

### Actual behavior

Running the application with `dotnet app.dll` results in a native `Aborted` crash.

I'm attaching corehost trace and a core dump below. I've instructed user how to generate those so it could aid debugging:

[corehost_trace.txt](https://github.com/dotnet/runtime/files/7654656/corehost_trace.txt)
[core.dump.zip](https://github.com/dotnet/runtime/files/7654661/core.dump.zip)

Both files were generated inside of Docker container.

### Regression?

Yes, the user has reported that the application worked fine with .NET 5.0.

### Known Workarounds

The user is able to run the application just fine on his host machine - this suggests a compatibility issue between Docker and .NET 6.0. However, .NET 5.0 app has worked fine in the same environment, and some other non-NET containers are also running without issues. Other users of the same app (me included) are unable to reproduce this issue, the app works fine in Docker for everybody else but this user.

### Configuration

Host machine:

```
Linux omv 5.10.63-v7l+ #1459 SMP Wed Oct 6 16:41:57 BST 2021 armv7l GNU/Linux
```

Docker version:
```
Client: Docker Engine - Community
 Version:           20.10.11
 API version:       1.41
 Go version:        go1.16.9
 Git commit:        dea9396
 Built:             Thu Nov 18 00:37:24 2021
 OS/Arch:           linux/arm
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.11
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.9
  Git commit:       847da18
  Built:            Thu Nov 18 00:35:41 2021
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Other information

This issue could be somehow related to https://github.com/dotnet/runtime/issues/62273 but I doubt it, as there the user could run the app which wasn't a self-contained single file app, while this one fails when running `dotnet` on framework-dependent one. Just for a reference, the same app with self-contained runtime also crashes in similar way - I've attached core dump and corehost trace from framework-dependent app crash to aid debugging.

However, both users are running my app on `linux-arm` machines, so it's possible that the same bug triggers in two different scenarios.

Please let me know what else I can provide/do to help narrow this one down, although I don't think I can offer much more than what I've already provided due to specifics of this bug. I had no luck reproducing it myself on any of my machines, but I strongly believe this is a regression in regards to .NET 5.0. Perhaps one of you will be able to reproduce this problem and therefore gather more info required to fix it.

Thank you in advance for your interest in regards to this issue.

**Comment by JustArchi** [comment]
BTW I've got the second user with exactly the same issue, on Raspberry Pi 4 32-bit (Raspberry Pi OS probably).

Uploading another trade and core dump seems unnecessary, but let me know if I should follow on that.

**Comment by pflannery** [comment]
I also hit this issue on my pi4 32bit debian 10 build 

```
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
```

I found this related issue https://giters.com/dotnet/dotnet-docker/issues/3253 
So looks like it's an issue related to an update to the clock on debian 10\11 32bit builds.

After reading that issue what I did was add the testing repo, update the offending package and then remove the testing repo.

**WARNING: do not attempt to run apt upgrade whilst the testing repo is listed as a source**

Steps:

1 Edit  ```/etc/apt/sources.list``` file and add the following line to include the testing repo as a source
  > deb http://raspbian.raspberrypi.org/raspbian/ testing main

2 Then run ```apt-get install libseccomp2```

3 Again edit  ```/etc/apt/sources.list``` file and remove the testing repo line you added in step 1

dotnet 6 now works as before.

**Comment by JustArchi** [comment]
I'd like to let you know that my user has confirmed the instructions above sent by @pflannery has fixed the issue for him as well, so indeed `libseccomp2` is at fault here.

Thanks a lot for a valid workaround :slightly_smiling_face:.

**Comment by KamilLenica** [comment]
This doesn't fix issue for me

> libseccomp2 is already the newest version (2.4.4-1~bpo10+1).

@pflannery can You check which version of libseccomp2 You are using?

**Comment by pflannery** [comment]
@KamilLenica 2.5.3-2+rpi1

**Comment by KamilLenica** [comment]
I've not performed 
> sudo apt update after editing sources.list, was that omitted intentionally, or shoud i run update before installing libseccomp2 ?

**Comment by pflannery** [comment]
@KamilLenica I didn't need to run update but my pi may of been up to date before I started "step 1".

**Comment by KamilLenica** [comment]
I've managed to install

>libseccomp2:armhf 2.5.3-2+rpi1

@JustArchi issue is still there

What might be impacting is that i have set 
> arm_64bit=1
on 32bit raspbian, which was needed for another container to run

**Comment by JustArchi** [comment]
@KamilLenica likely your docker is running a 64-bit process then, ensure you have `libseccomp2:arm64` in addition to `:armhf`.

**Comment by KamilLenica** [comment]
@JustArchi 
I don't see that available

`apt list -a libseccomp2
Listing... Done
libseccomp2/testing,now 2.5.3-2+rpi1 armhf [installed]
libseccomp2/oldstable 2.3.3-4 armhf`

guess i have to stay on older version of your app

**Comment by JustArchi** [comment]
Well I dunno then, what I'd recommend is actually installing a fully 64-bit OS like Debian, I have Raspberry Pi 4 running arm64 Debian Testing and I don't have this issue whatsoever.

**Comment by agocke** [comment]
It looks like this was resolved as a seccomp issue, or general configuration issue with 32-bit OS on AMR64, with an ARM64 process -- either way not likely a product issue. We've narrowed down the single-file issue and it seems extremely unlikely that code would have any effect here, it only runs for single-file.

I'll close for now, but if we get more evidence this is a product issue I can re-open and we can start a new investigation.
